File: backend/app/schemas/subscription_pause.py
from pydantic import BaseModel
from typing import Optional

class CreateSubscriptionPauseRequest(BaseModel):
    subscription_id: str
    start_date: str
    end_date: str
    reason: Optional[str] = None

class SubscriptionPauseResponse(BaseModel):
    id: str
    user_id: str
    subscription_id: str
    start_date: str
    end_date: str
    reason: Optional[str] = None
    created_at: Optional[str] = None

File: frontend/types/subscription-pause.ts
export interface SubscriptionPause {
  id: string;
  user_id: string;
  subscription_id: string;
  start_date: string;
  end_date: string;
  reason?: string;
  created_at: string;
}

export interface CreateSubscriptionPauseInput {
  subscription_id: string;
  start_date: string;
  end_date: string;
  reason?: string;
}
